includes/core/server/request/class-generic.php

```diff
@@ -0,0 +1,126 @@
+<?php
+/**
+ * Class file for WCPay\Core\Server\Request\Generic.
+ *
+ * @package WooCommerce Payments
+ */
+
+namespace WCPay\Core\Server\Request;
+
+use WCPay\Core\Exceptions\Invalid_Request_Parameter_Exception;
+use WCPay\Core\Exceptions\Server_Request_Exception;
+use WCPay\Core\Server\Request;
+use Requests;
+
+/**
+ * Generic WCPay Server Request.
+ *
+ * This class is not extendable, if you need something specific,
+ * it will not be generic anymore, create your own.
+ */
+final class Generic extends Request {
+	/**
+	 * The request's API.
+	 *
+	 * @var string Check WCPay\Core\Server\APIs.
+	 */
+	private $api;
+
+	/**
+	 * The method of the request.
+	 *
+	 * @var string See the constants in WordPress's `Requests` class.
+	 */
+	private $method;
+
+	/**
+	 * If true, the request will be signed with the user token rather than blog token. Defaults to false.
+	 *
+	 * @var bool
+	 */
+	private $should_use_user_token = false;
+
+	/**
+	 * Creates a new instance of the class.
+	 *
+	 * @throws Server_Request_Exception
+	 */
+	public static function create() {
+		throw new Server_Request_Exception( 'You cannot create request this way.', 'wcpay_core_server_request_invalid_method_call' );
+	}
```

Comment:
I don't understand this method. The parent class doesn't have such a method to override.

Reply:
The parent class used to have it until https://github.com/Automattic/woocommerce-payments/pull/5136/commits/49170a83303cf7a2e5d26d8e70104972f9e51dce#diff-8f66861195fea9a9c2c7473a1ab6691d71e2cff20214347eca77c728fd4ea65b, this is a left-over, which should be removed.